packages/vscode-extension/src/commonlib/telemetry.ts

```diff
@@ -177,4 +160,38 @@ export class VSCodeTelemetryReporter extends vscode.Disposable implements Teleme
   async dispose() {
     await this.reporter.dispose();
   }
+
+  addCommonProperties(properties: { [p: string]: string }) {
+    const rootPath = globalVariables.workspaceUri?.fsPath;
+    const projectSettings = getProjectSettingsForCommonTelemetry(rootPath);
+
+    if (
+      properties[TelemetryProperty.ProjectId] === "unknown" ||
+      properties[TelemetryProperty.ProjectId] === undefined
+    ) {
+      properties[TelemetryProperty.ProjectId] = projectSettings?.projectId
+        ? projectSettings?.projectId
+        : "unknown";
+    }
+
+    if (!properties[TelemetryProperty.IsFromSample]) {
```

Comment:
This is covered in `sharedProperties`. It will not be changed during the lifecycle of a project, so it's initialized once and not updated. Does it make sense to set dynamic properties in this method and move static ones to `sharedProperties`.